src/vendor/vendor.c

```diff
@@ -141,6 +145,47 @@ void rdk_free(void *ptr);
 #ifdef    INCLUDE_LCM_DATAMODEL
 #include "lcm_rbus_datamodel.c"
 #endif
+/*schedule_diagnostic to pass ConnectivityTestURL datamodel to RdkResetInner */
+int schedule_diagnostic(dm_req_t *req, char *command_key, kv_vector_t *input_args, kv_vector_t *output_args)
+{
+    int err = USP_ERR_OK;
+    USP_LOG_Info("Function called with command_key: %s\n", command_key);
+    err = RdkResetInner("pam", "Device.DeviceInfo.X_RDKCENTRAL-COM_Syndication.RDKB_UIBranding.NetworkDiagnosticTools.ConnectivityTestURL", "www.wikipedia.com", "DIAGNOSTIC");
+    USP_LOG_Info("schedule_diagnostic err = %d \n", err);
+    return err;
+}
+
+/* Command structure contains operate command and its respective function */
+typedef struct
+{
+    char *path;
+    dm_sync_oper_cb_t schedule;
+}Command;
+
+ Command commands[]=
+{
+    {"Device.Diagnostic()",schedule_diagnostic},
```

Comment:
Not sure on this command.whatever the command we implement.it should come from https://usp-data-models.broadband-forum.org/tr-181-2-14-0-usp.html#D.Device:2.Device.

Reply:
Hi Chandra,
we are adding support sample commands for operate option, to execute dmcli commands

For above command :  Device.Diagnostic() -> will perform set operation on **Device.DeviceInfo.X_RDKCENTRAL-COM_Syndication.RDKB_UIBranding.NetworkDiagnosticTools.ConnectivityTestURL** as **www.wikipedia.com**

Reply:
As updated in slack pls implement IPPing command